parser.js/ParserJS.idr

```diff
@@ -11,18 +11,26 @@ import Backend.ReasonML
 
 generateCode : String -> (n ** TNamed n) -> String
 generateCode "haskell"  (n  **tn) = toString $ generateDefs Haskell tn
-generateCode "reasonml" (n  **tn) = toString $ generateDefs ReasonML tn
-generateCode "json"     (Z  **tn) = toString $ generate JSONDef tn
-generateCode "json"     (S _**tn) = "<error : can't generate JSON schema for open typedef>"
-generateCode _          _         = "<error : unknown backend>"
+generateCode _          _         = "<error : unsupported backend>"
 
 -- re-exports
 parseType : String -> Maybe (n : Nat ** TNamed n)
 parseType = parseTNamed 
 
+generateSerializers : String -> String -> Maybe String
+generateSerializers backend tdef = map (generateCode backend) (parseType tdef)
+
+genType : String -> (n ** TNamed n) -> String
+genType "reasonml" (n   ** tn) = toString $ generateDefs ReasonML tn
+genType "json"     (Z   ** tn) = toString $ generate JSONDef tn
+genType "json"     (S _ ** tn) = "<error : cannot generte JSON schema for open typedefs>"
+genType _          _           = "<error : unsupported backend>"
+
+generateTypeSignature : String -> String -> Maybe String
```

Comment:
1) The naming difference between `genType` and `generateTypeSignature` seems a bit ad hoc; perhaps `generateTypeSignature'` would be 'better', instead of `genType`?
2) Re the naming of `generateSerializers` vs `generateTypeSignature`: The first seems to me to be a serialiser of terms, and the second a serialiser of types.
3) Also, isn't having `Signature` in there a bit superfluous?

Reply:
1: `genType` is private and only used as a helper for `generateTypeSignature` maybe I can move it as à where clause?

2-3: would renaming the second one as `generateType` or `generateTypeDefinition` or `generateDatatypes` be more helpful?

Reply:
1) `where` sounds good!

2) `generateType` is lovely, I think. 👍On top of that, don't you think having the word 'term' in the name `generateSerializers` somewhere would be good? Perhaps `generateTermSerializers`? It is a bit long, but for reasons of searchibility, it may be worth it. We could also do that in a later PR if we decide we want to. (And thinking about it again, my 'they are both serialisers' remark was a bit daft, so never mind that. `:P`)